shardingsphere-infra/shardingsphere-infra-executor/src/test/java/org/apache/shardingsphere/infra/executor/sql/federate/TestRuleConfiguration.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.executor.sql.federate;
+
+import org.apache.shardingsphere.infra.config.RuleConfiguration;
+
+public final class TestRuleConfiguration implements RuleConfiguration {
```

Comment:
Why do we need this class? I mean, shardingRule is not our target rule?

Reply:
Because we need a rule configuration when we initialize optimize context facotry.

Reply:
![image](https://user-images.githubusercontent.com/22066046/126067301-f84fa2db-b119-4dbc-82b5-f56fdc863173.png)